sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/ProtobufUtil.java

```diff
@@ -44,6 +39,43 @@ private ProtobufUtil() {
     }
   }
 
+  public static <MESSAGE extends Message> Iterable<MESSAGE> readFileMessages(File file, final Parser<MESSAGE> parser) {
+    try {
+      final InputStream inputStream = new BufferedInputStream(new FileInputStream(file));
+      return new Iterable<MESSAGE>() {
+        @Override
+        public Iterator<MESSAGE> iterator() {
+          return new Iterator<MESSAGE>() {
+
+            private MESSAGE currentMessage;
+
+            @Override
+            public boolean hasNext() {
+              try {
+                currentMessage = parser.parseDelimitedFrom(inputStream);
+              } catch (InvalidProtocolBufferException e) {
+                throw new IllegalStateException("Failed to read input stream", e);
+              }
+              return currentMessage != null;
+            }
+
+            @Override
+            public MESSAGE next() {
+              return currentMessage;
```

Comment:
In fact this design allowed me to implement the hasNext, otherwise I do not know how to do that...